components/eventing-controller/pkg/handlers/eventtype/clean_test.go

```diff
@@ -25,60 +26,65 @@ func TestCleaner(t *testing.T) {
 			givenApplicationName: "testapp",
 			givenEventType:       "sap.kyma.testapp.order.created.v1",
 			wantEventType:        "sap.kyma.testapp.order.created.v1",
+			wantError:            false,
 		},
 		{
 			givenEventTypePrefix:   "sap.kyma",
 			givenApplicationName:   "testapp",
 			givenApplicationLabels: map[string]string{application.TypeLabel: "testapptype"},
 			givenEventType:         "sap.kyma.testapp.order.created.v1",
 			wantEventType:          "sap.kyma.testapptype.order.created.v1",
+			wantError:              false,
 		},
 		{
 			givenEventTypePrefix: "sap.kyma",
 			givenApplicationName: "t..e--s__t!!a@@p##p%%",
 			givenEventType:       "sap.kyma.t..e--s__t!!a@@p##p%%.order.created.v1",
 			wantEventType:        "sap.kyma.testapp.order.created.v1",
+			wantError:            false,
 		},
 		{
 			givenEventTypePrefix:   "sap.kyma",
 			givenApplicationName:   "t..e--s__t!!a@@p##p%%",
 			givenApplicationLabels: map[string]string{application.TypeLabel: "t..e--s__t!!a@@p##p%%t^^y&&p**e"},
 			givenEventType:         "sap.kyma.t..e--s__t!!a@@p##p%%.order.created.v1",
 			wantEventType:          "sap.kyma.testapptype.order.created.v1",
+			wantError:              false,
 		},
 		// valid even-types but have not existing applications
 		{
 			givenEventTypePrefix: "sap.kyma",
 			givenApplicationName: "test-app",
 			givenEventType:       "sap.kyma.testapp.order.created.v1",
-			wantError:            true,
+			wantEventType:        "sap.kyma.testapp.order.created.v1",
+			wantError:            false,
 		},
 		{
 			givenEventTypePrefix: "sap.kyma",
 			givenApplicationName: "testapp",
 			givenEventType:       "sap.kyma.test-app.order.created.v1",
-			wantError:            true,
+			wantEventType:        "sap.kyma.testapp.order.created.v1",
+			wantError:            false,
 		},
 		// invalid even-types
 		{
 			givenEventTypePrefix: "sap.kyma",
-			givenApplicationName: "test-app",
-			givenEventType:       "sap.kyma.testapp.order.created.v1",
+			givenApplicationName: "testapp",
+			givenEventType:       "sap.kyma.testapp.order.created", // missing the version
 			wantError:            true,
 		},
 		{
 			givenEventTypePrefix: "sap.kyma",
 			givenApplicationName: "testapp",
-			givenEventType:       "sap.kyma.test-app.order.created.v1",
-			wantEventType:        "sap.kyma.testapp.order.created.v1",
+			givenEventType:       "sap.testapp.order.created.v1", // prefix not matched
```

Comment:
How about adding `name` for the testCase struct to provide some context and later use `t.Run("...context...", func(){})`?

Reply:
Done